core/src/Kokkos_SYCL.hpp

```diff
@@ -127,7 +127,6 @@ class SYCL {
     explicit SYCLDevice(sycl::device d);
     explicit SYCLDevice(const sycl::device_selector& selector);
     explicit SYCLDevice(size_t id);
-    explicit SYCLDevice(const std::function<bool(const sycl::device&)>& pred);
```

Comment:
What was the deal with this constructor?

Reply:
In https://github.com/kokkos/kokkos/pull/3231/files#diff-14e8d5edacfcbf256b2d86893b81b4ddb925537c1927d137eb16299f82353aacR109-R123, this constructor was implemented allowing passing a device selector (meaning a function taking a `sycl::device` to return `true` for the first acceptable `device` to be used). 

Currently, `SYCLDevice` is only used internally (similar to Cuda::SelectDevice`) and since it's not implemented I don't see a problem with backward compatibility.

Of course, we could also discuss in a separate pull request if we want to implement or remove this constructor (my preference is removing it).

Reply:
It was there to be able to provide a filter mechanism (and used in earlier versions when the runtime was buggier).  It can be removed.